drivers/sensor/adxl345/adxl345.c

```diff
@@ -0,0 +1,224 @@
+	int samples_count;
+	int rc;
+
+	data->sample_number = 0;
+	rc = adxl345_read_reg(dev, ADXL345_FIFO_STATUS_REG);
+	if (rc < 0) {
+		LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+		return rc;
+	}
+	samples_count = rc;
+	for (int s = 0; s < samples_count; s++) {
+		rc = adxl345_read_sample(dev, &sample);
+		if (rc < 0) {
+			LOG_ERR("Failed to fetch sample rc=%d\n", rc);
+			return rc;
+		}
+		data->bufx[s] = sample.x;
+		data->bufy[s] = sample.y;
+		data->bufz[s] = sample.z;
+	}
+
+	return samples_count;
+}
+
+static int adxl345_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+
+	if (data->sample_number > 32) {
+		data->sample_number = 0;
+	}
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		adxl345_accel_convert(val, data->bufx[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		adxl345_accel_convert(val, data->bufy[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		adxl345_accel_convert(val, data->bufz[data->sample_number]);
+		data->sample_number++;
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		adxl345_accel_convert(val++, data->bufx[data->sample_number]);
+		adxl345_accel_convert(val++, data->bufy[data->sample_number]);
+		adxl345_accel_convert(val,   data->bufz[data->sample_number]);
+		data->sample_number++;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api adxl345_api_funcs = {
+	.sample_fetch = adxl345_sample_fetch,
+	.channel_get = adxl345_channel_get,
+};
+
+static int adxl345_init(struct device *dev)
+{
+	int rc;
+	struct adxl345_dev_data *data = dev->driver_data;
+	const struct adxl345_dev_config *cfg = dev->config->config_info;
+
+	data->sample_number = 0;
+	data->i2c_master = device_get_binding(cfg->i2c_master_name);
+	data->i2c_addr = cfg->i2c_addr;
+
+	if (!data->i2c_master) {
+		LOG_ERR("Failed to get I2C master\n");
+		return -ENODEV;
+	}
+
+	rc = adxl345_read_reg(dev, ADXL345_DEVICE_ID_REG);
+	if (rc < 0 || rc != ADXL345_PART_ID) {
+		LOG_ERR("Read PART ID failed: 0x%x\n", rc);
+	}
+
+	rc = adxl345_write_reg(dev, ADXL345_FIFO_CTL_REG,
+			       ADXL345_FIFO_STREAM_MODE);
+	if (rc < 0) {
+		LOG_ERR("FIFO enable failed\n");
+	}
+
+	rc = adxl345_write_reg(dev, ADXL345_DATA_FORMAT_REG,
+			       ADXL345_RANGE_16G);
+	if (rc < 0) {
+		LOG_ERR("FIFO enable failed\n");
+	}
+
+	rc = adxl345_write_reg(dev, ADXL345_RATE_REG,
+			       ADXL345_RATE_25HZ);
+	if (rc < 0) {
+		LOG_ERR("Rate setting failed\n");
+	}
+
+	rc = adxl345_write_reg(dev, ADXL345_POWER_CTL_REG,
+			       ADXL345_ENABLE_MEASURE_BIT);
+	if (rc < 0) {
+		LOG_ERR("Enable measure bit failed\n");
+	}
+
+	return 0;
+}
+
+static struct adxl345_dev_data adxl345_data;
+
+static const struct adxl345_dev_config adxl345_config = {
+	.data = &adxl345_data,
+	.i2c_master_name = DT_INST_0_LITEX_I2C_LABEL,
```

Comment:
This should not be soc-specific. It should be `DT_INST_0_ADI_ADXL345_BUS_NAME` instead

Reply:
Done